libcaliptra/inc/caliptra_types.h

```diff
@@ -63,3 +63,117 @@ struct caliptra_stash_measurement_resp {
+    uint32_t                   data_size;
+    uint8_t                    data[1024];
+};
+
+struct caliptra_get_ldev_cert_resp {
+    struct caliptra_completion cpl;
+    uint32_t                   data_size;
+    uint8_t                    data[1024];
+};
+
+// The below fields are placeholders to set up the baseline
+// required for communication of DPE commands to Caliptra
+// firmware.
+
+#define DPE_DATA_MAX 512
+#define DPE_MAGIC    0x44504543 // "DPEC"
+
+struct dpe_cmd {
+    uint32_t magic;
+    uint32_t cmd_id;
+    uint32_t profile;
+};
+
+struct dpe_response {
+    uint32_t magic;
+    uint32_t status;
+    uint32_t profile;
+};
+
+struct dpe_get_profile_response {
+    struct dpe_response resp_hdr;
+    uint16_t major_version;
+    uint16_t minor_version;
+    uint32_t vendor_id;
+    uint32_t vendor_sku;
+    uint32_t max_tci_nodes;
+    uint32_t flags;
+};
+
+#define DPE_HANDLE_SIZE 16
+#define DPE_CERT_SIZE   2048
+
+#ifndef DPE_PROFILE
+#define DPE_PROFILE DPE_PROFILE_384
+#endif
+
+#if (DPE_PROFILE == DPE_PROFILE_256)
+#define DPE_ECC_SIZE 32
+#endif
+
+#if (DPE_PROFILE == DPE_PROFILE_384)
+#define DPE_ECC_SIZE 48
+#endif
+
+struct dpe_new_handle_response {
+    struct dpe_response resp_hdr;
+    uint8_t             handle[DPE_HANDLE_SIZE];
+};
+
+struct dpe_derive_child_response {
+    struct dpe_response resp_hdr;
+    uint8_t             handle[DPE_HANDLE_SIZE];
+    uint8_t             parent_handle[DPE_HANDLE_SIZE];
+};
+
+struct dpe_certify_key_response {
+    struct dpe_response resp_hdr;
+    uint8_t             new_context_handle[DPE_HANDLE_SIZE];
+    uint8_t             derived_pubkey_x[DPE_ECC_SIZE];
+    uint32_t            cert_size;
+    uint8_t             cert[DPE_CERT_SIZE];
+};
+
+struct dpe_sign_response {
+    struct dpe_response resp_hdr;
+    uint8_t             new_context_handle[DPE_HANDLE_SIZE];
+    uint8_t             sig_r_or_hmac[DPE_ECC_SIZE];
+    uint8_t             sig_s;
+};
+
+struct dpe_get_tagged_tci_response {
+    struct dpe_response resp_hdr;
+    uint8_t             tci_cumulative[DPE_ECC_SIZE];
+    uint8_t             tci_current[DPE_ECC_SIZE];
+};
+
+struct dpe_get_certificate_chain_response {
+    struct dpe_response resp_hdr;
+    uint32_t            certificate_size;
+    uint8_t             certificate_chain[DPE_CERT_SIZE];
+};
+
+struct caliptra_dpe_req {
+    caliptra_checksum checksum;
+    uint32_t          data_size;
+    uint8_t           data[DPE_DATA_MAX];
+};
+
+struct caliptra_dpe_resp {
+    struct caliptra_completion cpl;
+    uint32_t                   data_size;
+    union {
+        struct dpe_new_handle_response            new_handle;
+        struct dpe_derive_child_response          derive_child;
+        struct dpe_certify_key_response           certify_key;
+        struct dpe_sign_response                  sign;
+        struct dpe_get_tagged_tci_response        get_tagged_tci;
+        struct dpe_get_certificate_chain_response get_certified_chain;
```

Comment:
nit: Can you also include a union member that is `uint8_t data[DPE_RESP_MAX]` ([max size defined here](https://github.com/chipsalliance/caliptra-sw/blob/c76b3ed2bfbd8d6ee9b7183ac011288491c6b173/common/src/mailbox_api.rs#L245C9-L245C9))

This is helpful in cases where the caller to DPE is just proxying arbitrary commands on behalf of some other component. We will use this for calling the Golang DPE acceptance tests.

Reply:
Can do. Just for my knowledge, is there an equivalent in the Rust codebase or does it just handle it on the fly?

Reply:
In caliptra-dpe, the response struct implements as_bytes(), so callers can always interpret it as a slice if needed: https://github.com/chipsalliance/caliptra-dpe/blob/a28b6b701f232b9982ad5f5e0efd5f87293f5c5d/dpe/src/response.rs#L31

And the the invoke dpe mailbox command does this when writing to the DPE response structure: https://github.com/chipsalliance/caliptra-sw/blob/c76b3ed2bfbd8d6ee9b7183ac011288491c6b173/runtime/src/invoke_dpe.rs#L22